pink/barriers/barrier.py

```diff
@@ -172,15 +179,23 @@ def compute_qp_objective(
             Tuple containing the quadratic objective matrix (H) and linear
                 objective vector (c).
         """
-        jac = self.compute_jacobian(configuration)
+        if (
+            self.__q_cache is None
+            or self.__jac_cache is None
+            or self.__barrier_cache is None
+        ) or not np.allclose(self.__q_cache, configuration.q):
```

Comment:
The function `np.allclose` has relative/absolute tolerance parameters. Depending on the configuration space of the robot, users may want to set them to non-default values.

Reply:
I use `np.allclose` with default parameters with the hope that _in practice_ it would return False for any `q_cache != configuration.q`. Also see the discussion below for the logic behind caching.